tron/serialize/runstate/statemanager.py

```diff
@@ -201,6 +204,22 @@ def _save_from_buffer(self):
                 log.warning(msg)
                 raise PersistenceStoreError(msg)
 
+    def _check_consistency(self, state):
+        """ Consistency checks on serialized state to ensure no corruption.
+
+        :param state: A dict of serialized state, like that returned by
+            self.restore
+        """
+        log.info("Running consistency checks")
+        try:
+            # We attempt to save all the state. A failure indicates corruption
+            for type_enum, items in state.items():
+                for name, state_data in items.items():
+                    self.save(type_enum, name, state_data)
+        except PersistenceStoreError as e:
```

Comment:
Does this say what name/type/item/etc it couldn't save?

Reply:
I'm not sure. Like I said, I haven't been able to simulate the error. I only know that this exception is thrown based on that we try to handle it elsewhere when saves fail. @keymone do you know?

Reply:
the error happens when bdb file on disk is corrupt, you can't easily reproduce that in python. PersistenceStoreError won't have information about which key failed in this case - definitely worth try/except'ing inside the loop and re-raising with key name in the exception.

Reply:
We don't actually need to add the key name to the exception as it is already added (and logged) where we throw it (`_save_from_buffer`).